docs/source/java/overview.rst

```diff
@@ -0,0 +1,218 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: java
+.. highlight:: java
+
+===================
+High-Level Overview
+===================
+
+.. contents::
+
+Arrow Java modules is created using specification such as Columnar Format, Off-Heap
+Memory, Serialization and Inter Process Communication (IPC). Some of the Java modules
+was created with their own native implementations and others through bindings.
+
+Arrow Java provides several building blocks; ``data types`` describe the types of values;
+``ValueVectors`` are sequences of typed values; ``fields`` describe the types of columns in
+tabular data; ``schemas`` describe a sequence of columns in tabular data and
+``VectorSchemaRoot`` represents tabular data. Arrow also provides ``readers`` and
+``writers`` for loading data from and persisting data to storage.
+
+
+Arrow Java Modules
+------------------
+
+Arrow Java are divided in these modules to offer in-memory columnar data structures:
```

Comment:
I don't think the sentence adds anything

Reply:
Deleted